tests/models/test_circulationevent.py

```diff
@@ -190,7 +186,7 @@ def test_uniqueness_constraints_no_library(self):
 
         # Reuse the timestamp and you get an IntegrityError which ruins the
         # entire transaction.
-        assert_raises(
+        pytest.raises(
```

Comment:
Seems like if we're switching to pytest it makes sense to change these calls to the context manager form:

```Python
with pytest.raises(SomeException) as exc:
    ...
    call_that_raises_SomeException()
assert exc.value == "some message"
```

The old form isn't deprecated, but the docs do pretty strongly suggest going with the context manager. I'm also not sure you can 100% count on just subbing in `pytest.raises` for `assert_raises`. Maybe that's safe, but the declared signatures of the functions are different, so I'm not sure they always treat the contents of `*args` and `**kwargs` exactly the same.

Reply:
This construct also came from [nose2pytest](https://github.com/pytest-dev/nose2pytest). In their [documentation](https://github.com/pytest-dev/nose2pytest#status) they mention:
> Some Nose functions can be handled via a global search-replace, so a fixer was not a necessity:
    - assert_raises: replace with pytest.raises

So these were all replaced using a find / replace strategy. I did some basic testing, and it is working: the tests fail if the exception isn't raised or if a different exception is raised. 

Similar to my other comment, I'd like to keep this PR as a mostly automated update to pytest, and have a future effort to tackle some of the additional work that could be done fully utilizing pytest.

Reply:
Works for me!